rotkehlchen/tests/unit/test_uniswap.py

```diff
@@ -0,0 +1,380 @@
+import functools
+import pytest
+from unittest.mock import patch, MagicMock
+
+from rotkehlchen.assets.unknown_asset import UnknownEthereumToken
+from rotkehlchen.chain.ethereum.uniswap.typing import ProtocolBalance
+from rotkehlchen.chain.ethereum.uniswap.uniswap import Uniswap
+from rotkehlchen.constants import ZERO
+from rotkehlchen.fval import FVal
+from rotkehlchen.premium.premium import Premium
+from rotkehlchen.tests.utils.uniswap import (
+    ASSET_SHUF,
+    ASSET_TGX,
+    ASSET_USDT,
+    ASSET_WETH,
+    EXP_LIQUIDITY_POOL_1,
+    EXP_LIQUIDITY_POOL_2,
+    EXP_KNOWN_ASSETS_1,
+    EXP_KNOWN_ASSETS_2,
+    EXP_UNKNOWN_ASSETS_1,
+    EXP_UNKNOWN_ASSETS_2,
+    LIQUIDITY_POSITION_1,
+    LIQUIDITY_POSITION_2,
+    TEST_ADDRESS_1,
+    TEST_ADDRESS_2,
+    TEST_ADDRESS_3,
+    TOKEN_DAY_DATA_SHUF,
+    TOKEN_DAY_DATA_TGX,
+    UPDATED_LIQUIDITY_POOL_1,
+    UPDATED_LIQUIDITY_POOL_2,
+    UPDATED_LIQUIDITY_POOL_2_ONLY_USDT,
+    USD_PRICE_WETH,
+    USD_PRICE_USDT,
+    USD_PRICE_SHUF,
+    USD_PRICE_TGX,
+)
+from rotkehlchen.typing import Price
+
+
+@pytest.fixture
+def uniswap(
+        ethereum_manager,
+        database,
+        start_with_valid_premium,
+        rotki_premium_credentials,
+        function_scope_messages_aggregator,
+) -> Uniswap:
+    premium = None
+
+    if start_with_valid_premium:
+        premium = Premium(rotki_premium_credentials)
+
+    uniswap = Uniswap(
+        ethereum_manager=ethereum_manager,
+        database=database,
+        premium=premium,
+        msg_aggregator=function_scope_messages_aggregator,
+    )
+    return uniswap
+
+
+@pytest.fixture
+def patch_graph_query_limit(graph_query_limit):
+    with patch(
+        'rotkehlchen.chain.ethereum.uniswap.uniswap.GRAPH_QUERY_LIMIT',
+        new_callable=MagicMock(return_value=graph_query_limit),
+    ):
+        yield
+
+
+def store_call_args(func):
+    """
+    Helper function for pagination tests that call `self.graph.query()`.
+    """
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        call_args = {'args': args, 'kwargs': kwargs}
+        wrapper.calls.append(call_args)
+        return func(*args, **kwargs)
+
+    wrapper.calls = []
+    return wrapper
+
+
+class TestGetBalancesGraph:
```

Comment:
General comment on this test file. Nowhere in the code do we actually have tests inside classes. So if possible please don't do that here too. Just keep them out as normal functions like in the rest of the codebase